File: fastapi-app/schemas/content_extraction.py
from pydantic import BaseModel
from typing import Optional, Dict, Any
from datetime import datetime


class ExtractedContentResponse(BaseModel):
    id: int
    file_id: int
    content: Optional[str]
    metadata: Optional[Dict[str, Any]]
    extraction_status: str
    error_message: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SearchResult(BaseModel):
    file_id: int
    file_name: str
    content: Optional[str]
    score: float

    class Config:
        from_attributes = True


class SearchRequest(BaseModel):
    query: str
    limit: Optional[int] = 10
    offset: Optional[int] = 0


class SearchResponse(BaseModel):
    results: list[SearchResult]
    total: int
    query: str
    limit: int
    offset: int


class ExtractionTaskResponse(BaseModel):
    id: int
    file_id: int
    task_id: str
    status: str
    retry_count: int
    max_retries: int
    error_message: Optional[str]
    started_at: Optional[datetime]
    completed_at: Optional[datetime]
    created_at: datetime

    class Config:
        from_attributes = True


class ContentExtractionRequest(BaseModel):
    file_id: int
    force_reextract: bool = False

File: react-frontend/src/types/content-extraction.ts
export interface ExtractedContent {
  id: number;
  file_id: number;
  content: string | null;
  metadata: Record<string, unknown> | null;
  extraction_status: 'pending' | 'processing' | 'completed' | 'failed';
  error_message: string | null;
  created_at: string;
  updated_at: string;
}

export interface AITag {
  id: number;
  file_id: number;
  tag: string;
  confidence: number;
  source: 'spacy' | 'gliclass' | 'api';
  created_at: string;
}

export interface AISummary {
  id: number;
  file_id: number;
  summary: string;
  model: 'llama-3.1-storm-8b' | 'gpt-4' | 'claude-3' | 'deepseek';
  length: 'short' | 'medium' | 'long';
  created_at: string;
}

export interface SearchResult {
  file_id: number;
  file_name: string;
  content: string | null;
  score: number;
  highlights?: Array<{
    text: string;
    positions: Array<{ start: number; end: number }>;
  }>;
}

export interface SearchRequest {
  query: string;
  limit?: number;
  offset?: number;
  search_content?: boolean;
  tags?: string[];
}

export interface SemanticSearchResult {
  file_id: number;
  file_name: string;
  file_type: string;
  similarity_score: number;
  distance: number;
}

export interface SimilarFilesResponse {
  file_id: number;
  similar_files: SemanticSearchResult[];
  count: number;
}

export interface SearchResponse {
  results: SearchResult[];
  total: number;
  query: string;
  limit: number;
  offset: number;
  semantic?: boolean;
}

export interface ExtractionTask {
  id: number;
  file_id: number;
  task_id: string;
  status: 'pending' | 'processing' | 'completed' | 'failed';
  retry_count: number;
  max_retries: number;
  error_message: string | null;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
}

export interface ContentExtractionRequest {
  file_id: number;
  force_reextract?: boolean;
}

export interface ProcessingStatus {
  file_id: number;
  file_name: string;
  status: 'pending' | 'processing' | 'completed' | 'failed';
  progress?: number;
  error?: string;
  retry_count?: number;
}

export interface ContentPreviewData {
  file_name: string;
  content: string;
  metadata: Record<string, unknown>;
  extraction_status: string;
}

export interface AITagRequest {
  file_id: number;
  strategy: 'spacy' | 'gliclass' | 'api' | 'hybrid';
}

export interface AISummaryRequest {
  file_id: number;
  model: 'llama-3.1-storm-8b' | 'gpt-4' | 'claude-3' | 'deepseek';
  length: 'short' | 'medium' | 'long';
}
